pkg/clients/redis/redis.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package redis
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	sdkerrors "github.com/aliyun/alibaba-cloud-sdk-go/sdk/errors"
+	aliredis "github.com/aliyun/alibaba-cloud-sdk-go/services/r-kvstore"
+
+	"github.com/crossplane/provider-alibaba/apis/redis/v1alpha1"
+)
+
+var (
+	// ErrDBInstanceNotFound indicates DBInstance not found
+	ErrDBInstanceNotFound = errors.New("DBInstanceNotFound")
+)
+
+const (
+	// ErrCodeInstanceNotFound error code of ServerError when DBInstance not found
+	ErrCodeInstanceNotFound = "InvalidInstanceId.NotFound"
+	// DefaultReadTime indicates default connect timeout number
+	DefaultReadTime = 60 * time.Second
+	// PubilConnectionDomain indicates instances connect domain
+	PubilConnectionDomain = "-pb.redis.rds.aliyuncs.com"
+	// HTTPSScheme indicates request scheme
+	HTTPSScheme = "https"
+	// VPCNetworkType indicates network type by vpc
+	VPCNetworkType = "VPC"
+)
+
+// Client defines Redis client operations
+type Client interface {
+	DescribeDBInstance(id string) (*DBInstance, error)
+	CreateAccount(id, username, password string) error
+	CreateDBInstance(*CreateRedisInstanceRequest) (*DBInstance, error)
+	DeleteDBInstance(id string) error
+	AllocateInstancePublicConnection(id string, port int) (string, error)
+	ModifyDBInstanceConnectionString(id string, port int) (string, error)
+	Update(id string, req *ModifyRedisInstanceRequest) error
+}
+
+// DBInstance defines the DB instance information
+type DBInstance struct {
+	// Instance ID
+	ID string
+
+	// Instance status
+	Status string
+
+	// Endpoint specifies the connection endpoint.
+	Endpoint *v1alpha1.Endpoint
+}
+
+// CreateRedisInstanceRequest defines the request info to create DB Instance
+type CreateRedisInstanceRequest struct {
+	Name           string
+	InstanceType   string
+	EngineVersion  string
+	SecurityIPList string
+	InstanceClass  string
+	Password       string
+	ChargeType     string
+	Port           int
+	NetworkType    string
+	VpcID          string
+	VSwitchID      string
+}
```

Comment:
Is `EditionType` an important parameter for Redis? 
https://www.alibabacloud.com/help/doc-detail/60933.htm?spm=a2c63.p38356.b99.319.7c89614d0xGhX7

Reply:
it can be changed through instanceclass，createinstance does not have Editiontype field